### ISSUE TITLE
ExecutionFailed: Did not find any type information

### ISSUE THREAD

**Issue by CelineWuest** [neo4j/neo4j#13163]
When running the following query:
```cypher
RETURN 0 LIMIT (CASE WHEN true THEN 1 WHEN ((null>null) XOR null) THEN 1 END)
```

I encountered the following error:
```
Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (Did not find any type information for expression Null())
```

I believe the query mentioned above is semantically and syntactically correct and thus no error should be thrown here.

I encountered this issue when testing queries against the **Neo4j 5.6.0 enterprise version** in a Docker container running **alpine v.3**.

### Steps to reproduce
Run the following query and observe it throws an error:
```cypher
RETURN 0 LIMIT (CASE WHEN true THEN 1 WHEN ((null>null) XOR null) THEN 1 END)
```

### Expected behavior
The query should run successfully

### Actual behavior
The query fails with the error message `Neo4jError: Neo.DatabaseError.Statement.ExecutionFailed (Did not find any type information for expression Null())`

**Comment by loveleif** [comment]
Thanks @DominicWuest! Agree, I've informed the team ([private tracker link](https://trello.com/c/gHuYHB4m)).

**Comment by Lojjs** [comment]
@DominicWuest This was due to a bug in our type checking. It has now been fixed and the fix will be included in Neo4j 4.4.23 and 5.0.10. Thanks for reporting!
